### PR TITLE
Adding data connect devices and license dataset

### PR DESCRIPTION
**Pull request by rishicjain**

### CODE REVIEW

File: Datasets/data-connect-dataset-licensedetails.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## JSON representation
+
+```json
+{  
+"id": "String (identifier)",
+"skuId": "Guid",
+"skuPartNumber": "String"
+"servicePlans": [{"@odata.type": "microsoft.graph.servicePlanInfo"}]
+}
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-licensedetails.md:61]: Please add puser and ptenant in the json representation as well

File: Datasets/data-connect-dataset-licensedetails.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+| ptenant | string |  Tenant id. | No  |  None |
+
+
+
+## JSON representation
+
+```json
+{  
+"id": "String (identifier)",
+"skuId": "Guid",
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-licensedetails.md:58]: skuId should be "String (identifier)" just like id field and not "Guid"

File: Datasets/data-connect-dataset-owneddevices.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+| displayName | string | The display name for the device. | No | None |
+| domainName | string | The on-prem domain name of hybrid azure AD joined devices. This property is set by intune. | No | None |
+| enrollmentProfileName | string | Enrollment profile applied to the device. For example, apple device enrollment profile, Device enrollment - Corporate device identifilers, or Windows autopilot profile name. This property is set by intune. | No | None |
+| enrollmentType | object | Enrollment type of the device. This property is set by intune. Possible values are :- unkown, userEnrollment, deviceEnrollmentManager, appleBulkWithUser, appleBulkWithoutUser, windowsAzureADJoin, windowsBulkuserless, windowsAutoEnrollment, windowsBulkAzureDomainJoin, windowsCoManagement. | No | None |
+| extensionAttributes | array | Contains extension attributes 1-15 for the device. The individual extension attributes are not selectable. These properties are mastered in cloud and can be set during creation or update of a device object in Azure AD. FORMAT :- ARRAY<STRUCT<"extensionAttribute1":"String", "extensionAttribute2":"String", ... >> | No | None |
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-owneddevices.md:55]: I see the json code shows this field as object: https://dev.azure.com/O365Exchange/O365%20Core/_git/MARS?path=/sources/dev/mars/src/DiscoveryService/Data/BasicDataSet_v0_OwnedDevices_v0.json&version=GBmaster&line=79&lineEnd=79&lineStartColumn=18&lineEndColumn=24&lineStyle=plain&_a=contents

File: Datasets/data-connect-dataset-owneddevices.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+| deviceId | string | Unique identifier set by Azure Device Registration Service at the time of registration. This is an alternate key that can be used to refrence the device object. | No | None |
+| deviceOwnership | string | Ownership of the device. This property is set by intune. Possible values are :- unkown, company, personal. | No | None |
+| displayName | string | The display name for the device. | No | None |
+| domainName | string | The on-prem domain name of hybrid azure AD joined devices. This property is set by intune. | No | None |
+| enrollmentProfileName | string | Enrollment profile applied to the device. For example, apple device enrollment profile, Device enrollment - Corporate device identifilers, or Windows autopilot profile name. This property is set by intune. | No | None |
+| enrollmentType | object | Enrollment type of the device. This property is set by intune. Possible values are :- unkown, userEnrollment, deviceEnrollmentManager, appleBulkWithUser, appleBulkWithoutUser, windowsAzureADJoin, windowsBulkuserless, windowsAutoEnrollment, windowsBulkAzureDomainJoin, windowsCoManagement. | No | None |
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-owneddevices.md:54]: I see this is string type in json code: https://dev.azure.com/O365Exchange/O365%20Core/_git/MARS?path=/sources/dev/mars/src/DiscoveryService/Data/BasicDataSet_v0_OwnedDevices_v0.json&version=GBmaster&line=71&lineEnd=71&lineStartColumn=18&lineEndColumn=24&lineStyle=plain&_a=contents

File: Datasets/data-connect-dataset-registereddevices.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+| deviceId | string | Unique identifier set by Azure Device Registration Service at the time of registration. This is an alternate key that can be used to refrence the device object. | No | None |
+| deviceOwnership | string | Ownership of the device. This property is set by intune. Possible values are :- unkown, company, personal. | No | None |
+| displayName | string | The display name for the device. | No | None |
+| domainName | string | The on-prem domain name of hybrid azure AD joined devices. This property is set by intune. | No | None |
+| enrollmentProfileName | string | Enrollment profile applied to the device. For example, apple device enrollment profile, Device enrollment - Corporate device identifilers, or Windows autopilot profile name. This property is set by intune. | No | None |
+| enrollmentType | object | Enrollment type of the device. This property is set by intune. Possible values are :- unkown, userEnrollment, deviceEnrollmentManager, appleBulkWithUser, appleBulkWithoutUser, windowsAzureADJoin, windowsBulkuserless, windowsAutoEnrollment, windowsBulkAzureDomainJoin, windowsCoManagement. | No | None |
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-registereddevices.md:54]: I see this is string in json code: https://dev.azure.com/O365Exchange/O365%20Core/_git/MARS?path=/sources/dev/mars/src/DiscoveryService/Data/BasicDataSet_v0_RegisteredDevices_v0.json&version=GBmaster&line=71&lineEnd=71&lineStartColumn=18&lineEndColumn=24&lineStyle=plain&_a=contents

File: Datasets/data-connect-dataset-registereddevices.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+| extensionAttributes | array | Contains extension attributes 1-15 for the device. The individual extension attributes are not selectable. These properties are mastered in cloud and can be set during creation or update of a device object in Azure AD. FORMAT :- ARRAY<STRUCT<"extensionAttribute1":"String", "extensionAttribute2":"String", ... >> | No | None |
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-registereddevices.md:55]: I see this is object in json code: https://dev.azure.com/O365Exchange/O365%20Core/_git/MARS?path=/sources/dev/mars/src/DiscoveryService/Data/BasicDataSet_v0_RegisteredDevices_v0.json&version=GBmaster&line=79&lineEnd=79&lineStartColumn=18&lineEndColumn=24&lineStyle=plain&_a=contents

File: Datasets/data-connect-dataset-registereddevices.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
...mainName":null,"enrollmentProfileName":null,"enrollmentType":null,"id":"de217b7e-0c67-4ca2-b5b7-868f506090ed","isCompliant":null,"isManaged":null,"isRooted":null,"managementType":null,"manufacturer":null,"mdmAppId":null,"model":null,"onPremisesLastSyncDateTime":null,"onPremisesSyncEnabled":null,"operatingSystem":"Windows","operatingSystemVersion":"10.0.22581.200","profileType":"RegisteredDevice","registrationDateTime":"2022-04-06T22:00:54Z","systemLabels":[],"trustType":"Workplace","extensionAttributes":{"extensionAttribute1":null,"extensionAttribute2":null,"extensionAttribute3":null,"extensionAttribute4":null,"extensionAttribute5":null,"extensionAttribute6":null,"extensionAttribute7":null,"extensionAttribute8":null,"extensionAttribute9":null,"extensionAttribute10":null,"extensionAttribute11":null,"extensionAttribute12":null,"extensionAttribute13":null,"extensionAttribute14":null,"extensionAttribute15":null},"puser":"X530bf91-e844-4369-a808-e0d12b1008cd","ptenant":"Xe56195d-f07c-44f0-8108-40e4352e3e74","pAdditionalInfo":null,"datarow":0,"userrow":0,"pagerow":0,"rowinformation":{"errorInformation":null,"userReturnedNoData":false,"isUserSummaryRow":false,"userHasCompleteData":false}}
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-registereddevices.md:115]: Please make sure in all 3 datasets, there is no PII data in samples. Please recheck all sample records

File: Datasets/data-connect-dataset-registereddevices.md (hunk 6 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-registereddevices.md]: Resolved in latest commit. This was by mistake it is object in json as well as sample data from sample runs.

File: Datasets/data-connect-dataset-registereddevices.md (hunk 5 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-registereddevices.md]: Missed changing type while copying table rows. Fixed in latest commit

File: Datasets/data-connect-dataset-owneddevices.md (hunk 3 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-owneddevices.md]: Resolved in latest commit. This was by mistake it is object in json as well as sample data from sample runs.

File: Datasets/data-connect-dataset-owneddevices.md (hunk 4 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-owneddevices.md]: Missed changing type while copying table rows. Fixed in latest commit

File: Datasets/data-connect-dataset-licensedetails.md (hunk 2 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-licensedetails.md]: Resolved

File: Datasets/data-connect-dataset-licensedetails.md (hunk 1 above)
Comment (rishicjain) [reply Datasets/data-connect-dataset-licensedetails.md:63]: Added

File: Datasets/data-connect-dataset-licensedetails.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+## Definitions
+
+- Licenses are assigned for every purchase of any microsoft product/service(eg: M365) by a tenant. They represent the users/groups allowed to use products within a company.  
+
+## Schema
+
+| Name  | Type  |  Description  |  FilterOptions  |  FilterType  |
+| ----------- | ----------- | ----------- | ----------- | ----------- |
+| id | string | The unique identifier of the license details object | No | None |
+| skuID | string | Unique Identifier(GUID) for the service SKU. Equal to the skuID property on the related SubscribedSKU object. | No | None |
```
Comment (pushkarprotik) [inline Datasets/data-connect-dataset-licensedetails.md:44]: Is this field: "skuId" or "skuID"? Sample and Json representation say skuId

Comment (rishicjain) [reply Datasets/data-connect-dataset-licensedetails.md:44]: Just checked even the sample runs are giving skuId ... Changed in the latest commit

File: Datasets/data-connect-dataset-licensedetails.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+| skuPartNumber | string | Unique SKU display name. Equal to the skuPartNumber on the related SubscribedSku object. | No | None |
+| servicePlans | array | Information about the service plans assinged with the license. FORMAT:- ARRAY<STRUCT<'servicePlanId':'String', 'servicePlanName': 'String', 'provisioningStatus':'String', 'appliesTo':'string'>> | No | None |
+| originalStart | datetime | The start time that was set when the event was created in UTC time. | No | None |
```
[the remark this answers is not in the document]
Comment (rishicjain) [reply Datasets/data-connect-dataset-licensedetails.md:47]: This is copied from the schema karan had shared. I believe there are mistakenly copied rows there. Removing it in the latest PR:- as even the docs dont mention this field.

**Pull request closed without merging**